backend/modules/instruments/services.py

```
@@ -16,9 +19,10 @@ class PaginatedInstruments(TypedDict):
     num_pages: int
 
 
-class InstrumentsServiceMinimal:
-    def __init__(self):
-        self._repository = YfinanceRepository()
+class InstrumentsService:
+    def __init__(self, repository: InstrumentsRepository | None = None):
+        logger.debug("Initializing InstrumentsService with repository: %s", repository)
+        self._repository = repository or InstrumentsRepository()
```

Pylint (3.3.7)

[error] No value for argument 'available_instruments' in constructor call

(E1120)